gnovm/pkg/gnolang/preprocess.go

```diff
@@ -13,11 +13,21 @@ import (
 // Anything predefined or preprocessed here get skipped during the Preprocess
 // phase.
 func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
+	for _, fn := range fset.Files {
```

Comment:
This solution only works on the GnoVM by working on the individual file; it doesn't take into consideration dependencies across files.

It doesn't attempt to rewrite/adapt the existing code of the GnoVM which is in charge of dependency initialization (even by removing it, which would be fine, but we shouldn't have dead code responsible for doing the same thing.) I'm referring to recursive initialization through `predefineNow` / `tryPredefine`.

To explain what I mean by the first problem, I split up `var21.gno` in two files (a.gno and b.gno) and the initialization is still failing:

```go
// a.gno
package main

func main() {
	myDep = "123"
	myVar()
}

func hello(s string) {
	println(s)
}

var myVar = func() { hello(myDep) }
```

```go
// b.gno
package main

var myDep string
```

This still panics:

```
$ gnocco run .
panic: name myDep not declared [recovered]
        panic: main/a.gno:12: name myDep not declared:
```